bridges/modules/messages/src/lib.rs

```diff
@@ -640,18 +640,16 @@ pub mod pallet {
 
 				// collect message_nonces that were supposed to be pruned
 				let mut unpruned_message_nonces = Vec::new();
-				let mut nonce_to_check = expected_last_prunned_nonce;
-				loop {
+				const MAX_MESSAGES_ITERATION: u64 = 16;
+				let start_nonce =
+					expected_last_prunned_nonce.checked_sub(MAX_MESSAGES_ITERATION).unwrap_or(0);
+				for current_nonce in start_nonce..=expected_last_prunned_nonce {
```

Comment:
I'm not familiar with `try_state()`. Why do we use `MAX_MESSAGES_ITERATION`  and not iterate all the way ? 

Otherwise replacing the `loop` wit ha `for` looks good.

Reply:
> I'm not familiar with `try_state()`. Why do we use `MAX_MESSAGES_ITERATION` and not iterate all the way ?

Because, on every runtime upgrade it would be more and more waste of time, e.g.
- first time it would run from e.g. message with nonce 1000 to 0
- another runtime upgrade (when more messages happen) it would run e.g. 3000 to 0 (we previously checked interval 1000 to 0)
- another runtime upgrade (when more messages happen) it would run e.g. 6000 to 0 (we previously checked interval 3000 to 0)
- and so on

Yes, that's why I updated description that maybe `oldest_unpruned_nonce` and `do_try_state_for_outbound_lanes` are not really needed.

Slava wanted to create migration, that remove all messages < `oldest_unpruned_nonce`, but I think his concert was valid only if we have already more than one lane, because `on_idle` he picked different/random lanes to prune depending on `ActiveOutboundLanes::count`, which is for fellows 1 :), I think MBM migration is too much for this and also for standard migration, we would need to handle consumed weights and store last pruned block somewhere else, because `oldest_unpruned_nonce` is incremented on delivery now.

But as I am thinking about it again now, I think we should really remove attribute `lane.oldest_unpruned_nonce`, because we prune on delivery and this attribute does not make sense. So, the removal of this attribute could cause storage version change and migration, which would clean everything :)

Reply:
actually, MBM migration and removed `oldest_unpruned_nonce`, would do the job :)

Reply:
we don't use `oldest_unpruned_nonce` for relayer, anyway we can leave also without these migration, the worst can happen that we will have just a few messages in the storage, which we don't need. We add monitoring or do some sanity checks with seaching storage manually after upgrade, and when few messages left, we could remove them with `set_prefix` / `kill_storage`.